cloudify_agent/operations.py

```diff
@@ -446,3 +451,62 @@ def validate_agent_amqp(current_amqp=True, **_):
         raise NonRecoverableError(result['agent_alive_error'])
     if not current_amqp and not result['agent_alive_crossbroker']:
         raise NonRecoverableError(result['agent_alive_crossbroker_error'])
+
+
+@operation
+def transfer_agent_amqp(transfer_agent_timeout=300,
+                        manager_ip=None, manager_certificate=None,
+                        manager_rest_token=None, **_):
+
+    _create_broker_config(transfer_mode=True)
+    old_agent = _validate_agent()
+    agents = _run_install_script(old_agent, transfer_agent_timeout, manager_ip,
+                                 manager_certificate, manager_rest_token,
+                                 transfer_agent=True)
+    returned_agent = agents['new']
+    ctx.logger.info('Configured agent {0} to work with the new Manager'.
+                    format(returned_agent['name']))
+
+    # Make sure the agent is alive:
+    app = get_celery_app(tenant=returned_agent['rest_tenant'])
+    _assert_agent_alive(returned_agent['name'], app)
+
+
+def _create_broker_config(transfer_mode=False):
```

Comment:
If you aren't updating the runtime properties here, then why save it in them? Maybe just return a dict instead? If not, make sure to run `ctx.instance.update()` at the end of the function